sdk/core/Azure.Core.Experimental/src/CloudMachine.cs

```diff
@@ -0,0 +1,253 @@
+        /// <returns></returns>
+        /// <remarks>DisplayName is initialized to id. It can be changed by setting the DisplayName property.</remarks>
+        public static CloudMachine Create(string subscriptionId, string region)
+        {
+            if (string.IsNullOrEmpty(subscriptionId)) throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrEmpty(region)) throw new ArgumentNullException(nameof(region));
+
+            var id = GenerateCloudMachineId();
+            var defaultDisplayName = $"{id}@{DateTime.UtcNow:d}";
+            var cm = new CloudMachine(id, defaultDisplayName, subscriptionId, region, 1);
+            return cm;
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from configurationFile
+        /// </summary>
+        /// <param name="configurationFile"></param>
+        /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
+        public CloudMachine(string configurationFile = ".\\cloudconfig.json")
+        {
+            try
+            {
+                byte[] configurationContent = File.ReadAllBytes(configurationFile);
+                var document = JsonDocument.Parse(configurationContent);
+                JsonElement json = document.RootElement.GetProperty("CloudMachine");
+
+                Id = ReaderString(json, "id", configurationFile);
+                SubscriptionId = ReaderString(json, "subscriptionId", configurationFile);
+                Region = ReaderString(json, "region", configurationFile);
+                DisplayName = ReaderString(json, "name", configurationFile);
+                Version = ReaderInt32(json, "version", configurationFile);
+            }
+            catch (InvalidCloudMachineConfigurationException)
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new InvalidCloudMachineConfigurationException(configurationFile, setting: null, e);
+            }
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from stream.
+        /// </summary>
+        /// <param name="configurationContent"></param>
+        /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
+        public CloudMachine(Stream configurationContent)
+        {
+            try
+            {
+                var document = JsonDocument.Parse(configurationContent);
+                JsonElement json = document.RootElement.GetProperty("CloudMachine");
+
+                Id = ReaderString(json, "id", nameof(configurationContent));
+                SubscriptionId = ReaderString(json, "subscriptionId", nameof(configurationContent));
+                Region = ReaderString(json, "region", nameof(configurationContent));
+                DisplayName = ReaderString(json, "name", nameof(configurationContent));
+                Version = ReaderInt32(json, "version", nameof(configurationContent));
```

Comment:
nit: refactor duplicate code to a helper method?

Reply:
I tried. I don't know how. The ctor keeps claiming that fields are not initialized. Is there some nullability annotation I can use?